### ISSUE TITLE
%matplotlib notebook does not work

### ISSUE THREAD

**Issue by rodrigob** [nteract/hydrogen#782]
**Description:**

Using 
%matplotlib notebook
raises 
```
EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".
```

**Steps to Reproduce:**

1. Create a python document
2. Add `%matplotlib notebook`
3. Add code to plot anything
4. Run

**Versions:**

Which OS and which version of Hydrogen and Atom are you running?

Linux.
Atom    : 1.16.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0

**Comment by rodrigob** [comment]
Maybe related to #747 ?

**Comment by nikitakit** [comment]
I just want to mention that `%matplotlib inline` does work, even though it's not as nice as `%matplotlib notebook`.

Hopefully someone else can chime in on whether it's possible to get the notebook rendering working or not. As I understand it the issue here is that the notebook-rendered graphs use Javascript for interactivity, and Atom refuses to execute it without proper sandboxing from the text editor itself.

**Comment by rodrigob** [comment]
Thanks @nikitakit for the reply.
Yes I am using the `inline` mode, but I am looking forward to have interactive elements too.
I guess another work around in the mean time is to have a window pop-up (non-inline backend) for the piece of interactive code.

**Comment by lgeiger** [comment]
@rodrigob Thanks for the issue `%matplotlib notebook` would be a cool feature! 

Though unfortunately there are multiple problems why at the moment we don't support `%matplotlib notebook` out of the box:

1. Atom's Content Security Policy doesn't allow injection of scripts. We track this in #27 
2. https://github.com/nteract/nteract/issues/1359

For interactive plotting we only support plotly at the moment: https://nteract.gitbooks.io/hydrogen/docs/Usage/Examples.html#interactive-plots-using-plotly

I'm going to close this issue. We can continue the discussion in https://github.com/nteract/nteract/issues/1359

**Comment by lgeiger** [comment]
After a bit of googling I discovered a awesome functionality of plotly. It can be used to render matplotlib figures:

```python
import numpy as np
import matplotlib.pyplot as plt
from plotly import offline as py
import plotly.tools as tls
py.init_notebook_mode()

t = np.linspace(0, 20, 500)
plt.plot(t, np.sin(t))

py.iplot(tls.mpl_to_plotly(plt.gcf()))
```

This means one can render very nice interactive matplotlib figures inside Hydrogen. 🎉 

We should definitely add this to our Examples.

**Comment by kiwi0fruit** [comment]
@lgeiger I was not able to get plotly render 3D plots.

* I used [this manual](https://www.reddit.com/r/IPython/comments/3tibc8/tip_on_how_to_run_plotly_examples_in_offline_mode/). But for some reason 3D plots were not showing - only the area and controls. When I scroll down so the plot is not visible, then click on text, then scroll up - the plot appears. But it dissappears when I scroll up more or move mouse over the plot. Very strange and suspicious.
* I also tryed to pass matplotlib 3D plot to the plotly but this gave me the mistake (like I was sending something empy).
* So I dropped plotly as not working out of the box and found really nice solution:
* Using matplotlib with Qt. The code should be slightly modified:

```py
import matplotlib
matplotlib.use('Qt5Agg')  # this should be done before import matplotlib.pyplot, use 'QT4Agg' for older version
import matplotlib.pyplot as plt
```

This gave me interactive 3D plots that can be rotated right in the Hydrogen (actually in the separate window but who cares?). **So I suggest adding it to the [manual](https://nteract.gitbooks.io/hydrogen/docs/Usage/Examples.html#interactive-plots-using-plotly)**.
